proto/substrait/extended_expression.proto

```diff
@@ -19,33 +19,40 @@ message ExpressionReference {
     AggregateFunction measure = 2;
   }
   // Field names in depth-first order
+  //
+  // For primitive types, this will just be a single name. Nested types may
+  // have multiple names.
   repeated string output_names = 3;
 }
 
-// Describe a set of operations to complete.
-// For compactness sake, identifiers are normalized at the plan level.
+// Describe a list of expressions to evaluate.
+// This is a top-level expression, like a Plan.
 message ExtendedExpression {
   // Substrait version of the expression. Optional up to 0.17.0, required for later
   // versions.
   Version version = 7;
 
-  // a list of yaml specifications this expression may depend on
+  // A list of yaml specifications this expression may depend on.
   repeated substrait.extensions.SimpleExtensionURI extension_uris = 1;
 
-  // a list of extensions this expression may depend on
+  // A list of extensions this expression may depend on.
   repeated substrait.extensions.SimpleExtensionDeclaration extensions = 2;
 
-  // one or more expression trees with same order in plan rel
+  // One or more expression trees with same order in plan rel.
+  //
+  // The output schema is determined by these expression trees.
```

Comment:
In what way?  Is the output schema a single struct array with one field per expression tree?

Reply:
```suggestion
  // The output schema is determined by these expression trees. Each expression
  // tree becomes a field in the output schema. The order of the trees determines
  // the order of the output fields.
```